test/ReadsUtils_server_test.py

```diff
@@ -915,6 +915,211 @@ def test_interleaved_with_pe_inputs(self):
+        singleResultsNode = singleResultsD['lib']['file']['id']
+        self.delete_shock_node(node)
+        self.delete_shock_node(resultsNode)
+        self.delete_shock_node(singleResultsNode)
+
+    def test_wrong_obj_as_input(self):
+        # GET initial object in.
+        # First load source single ends reads file.
+        tf = 'Sample1.fastq'
+        target = os.path.join(self.scratch, tf)
+        shutil.copy('data/' + tf, target)
+        bad_object_type = {'type': 'Empty.AType',
+                           'data': {"foo": 3},
+                           'name': "bad_object"
+                           }
+        bad_object = self.dfu.save_objects({'id': self.ws_info[0],
+                                            'objects':
+                                            [bad_object_type]})[0]
+
+        bad_object_ref = str(bad_object[6]) + '/' + str(bad_object[0]) + \
+            '/' + str(bad_object[4])
+        # Now try upload of reads with ref to wrong object
+        self.fail_upload_reads(
+            {'fwd_file': target,
+             'wsname': self.ws_info[1],
+             'source_reads_ref': bad_object_ref,
+             'name': 'foo'
+             },
+            ("Invalid type for object {} (bad_object). Supported types: " +
+             "KBaseFile.SingleEndLibrary KBaseFile.PairedEndLibrary " +
+             "KBaseAssembly.SingleEndLibrary " +
+             "KBaseAssembly.PairedEndLibrary").format(bad_object_ref))
+
+    def test_single_end_obj_to_paired_end_error(self):
+        # GET initial object in.
+        # First load source single ends reads file.
+        tf = 'Sample1.fastq'
+        target = os.path.join(self.scratch, tf)
+        shutil.copy('data/' + tf, target)
+        ref = self.impl.upload_reads(
+            self.ctx, {'fwd_file': target,
+                       'sequencing_tech': 'illumina',
+                       'wsname': self.ws_info[1],
+                       'single_genome': 0,
+                       'name': 'fileReadsSingleSource'})
+        obj = self.dfu.get_objects(
+            {'object_refs': [self.ws_info[1] + '/fileReadsSingleSource']})['data'][0]
+        self.assertEqual(ref[0]['obj_ref'], self.make_ref(obj['info']))
+        self.assertEqual(obj['info'][2].startswith(
+                        'KBaseFile.SingleEndLibrary'), True)
+        d = obj['data']
+        node = d['lib']['file']['id']
+        # Now upload fail a paired end upload with a single end source
+        self.fail_upload_reads(
+            {'fwd_file': target,
+             'interleaved': 1,
+             'wsname': self.ws_info[1],
+             'source_reads_ref': ref[0]['obj_ref'],
+             'name': 'foo'
+             },
+            ("The input reference reads is single end, that should not " +
+             "give rise to a paired end object."))
+        self.delete_shock_node(node)
+
     def check_lib(self, lib, size, filename, md5):
```

Comment:
Needs a test to exercise the try / catch around fetching the source ref object with DFU, other than that looks good.